themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,548 @@
+lambda_target = aws.cloudwatch.EventTarget("lambda-target",
+    arn=lambda_function.arn,
+    rule=rule.name,
+    event_bus_name=bus.name,
+);
+
+# Give EventBridge permission to invoke the function.
+lambda_permission = aws.lambda_.Permission("lambda-permission",
+    action="lambda:InvokeFunction",
+    principal="events.amazonaws.com",
+    function=lambda_function.arn,
+    source_arn=rule.arn,
+);
+
+# Export the API Gateway URL to give us something to POST to.
+pulumi.export("url", pulumi.Output.concat(api.api_endpoint, "/", stage.name))
+```
+
+{{% /choosable %}}
+
+## All Together, Now
+
+Now that the program's complete, you can run Pulumi to bring it to life:
+
+```bash
+$ pulumi up
+...
+
+Updating (dev)
+...
+
+     Type                             Name                      Status
+ +   pulumi:pulumi:Stack              eventbridge-v2-dev        created
+ +   ├─ aws:apigatewayv2:Api          api                       created
+ +   ├─ aws:apigatewayv2:Stage        stage                     created
+ +   ├─ aws:cloudwatch:EventBus       bus                       created
+ ...
+
+Outputs:
+    apiURL: "https://geqfietbcl.execute-api.us-west-2.amazonaws.com/dev"
+
+Resources:
+    + 15 created
+
+Duration: 31s
+```
+
+When the update finishes, you'll have a fully functioning API Gateway-EventBridge integration that you can verify with `curl`:
+
+```bash
+$ curl --data '{"some-key": "some-value"}' --header "Content-Type: application/json" \
+     "$(pulumi stack output url)/uploads"
+
+{"Entries":[{"EventId":"cdc44763-6976-286c-9378-7cce674dff81"}],"FailedEntryCount":0}
+```
+
+Note the response, which comes directly from EventBridge (courtesy of the `AWS_PROXY` integration), confirming the event was received and written to the event bus.
+
+The final step is to confirm that the request made it all the way to Lambda, which you can easily do by tailing its output with `pulumi logs`:
+
+```bash
+$ pulumi logs --follow
+
+Collecting logs for stack dev since 2022-01-06T16:18:48.000-08:00.
+...
+
+{
+    source: 'my-event-source',
+    detail: { 'some-key': 'some-value' }
+}
+```
+
+When you're happy, be sure to tidy up with a `pulumi destroy`.
+
+## What's next?
+
+There's a lot more you can do with integrations like this that didn't cover: add more Lambda function handlers to the program, have EventBridge target [other AWS services](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-targets.html) ([Step Functions](https://aws.amazon.com/step-functions) might be a good one to try next), validate HTTP request bodies (with API Gateway [models](https://docs.aws.amazon.com/apigateway/latest/developerguide/models-mappings.html), to keep bad data from ever reaching EventBridge), and more. Hopefully this gives you sense of what's possible, though. And as promised, you'll find examples using both API Gateway V1 and V2 at [pulumi/examples](https://github.com:pulumi/example/FIXME).
```

Comment:
```suggestion
There's a lot more you can do with integrations like this that didn't cover: add more Lambda function handlers to the program, have EventBridge target [other AWS services](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-targets.html) ([Step Functions](https://aws.amazon.com/step-functions) might be a good one to try next), validate HTTP request bodies (with API Gateway [models](https://docs.aws.amazon.com/apigateway/latest/developerguide/models-mappings.html), to keep bad data from ever reaching EventBridge), and more. Hopefully this gives you sense of what's possible, though. And as promised, you'll find examples using both API Gateway V1 and V2 at [pulumi/examples](https://github.com/pulumi/examples) on GitHub.
```

Reply:
Yep, I'll update this link when the examples are merged. Good eye!